lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,177 @@
+        packaging_data = build_details[packaging_platform]
+        logger.debug("- #{ packaging_data }, isnil? #{ packaging_data.nil? }")
+        if packaging_data.nil?
+          message = <<-EOF
+            :packaging_platform '#{packaging_platform}' for host '#{host}' not in build details
+            :packaging_platform should be the platform-specific key from this list:
+              #{ build_details.keys }
+          EOF
+          fail_test( message )
+        end
+
+        artifact_buildserver_path   = packaging_data[:artifact]
+        repoconfig_buildserver_path = packaging_data[:repo_config]
+        fail_test('no artifact_buildserver_path found') if artifact_buildserver_path.nil?
+
+        artifact_url    = "#{build_url}/#{artifact_buildserver_path}"
+        repoconfig_url  = "#{build_url}/#{repoconfig_buildserver_path}" unless repoconfig_buildserver_path.nil?
+        artifact_url_correct = link_exists?( artifact_url )
+        logger.debug("- artifact url: '#{artifact_url}'. Exists? #{artifact_url_correct}")
+        fail_test('artifact url built incorrectly') if !artifact_url_correct
+
+        return artifact_url, repoconfig_url
+      end
+
+      # install build artifact on the given host
+      #
+      # @param [Host] host Host to install artifact on
+      # @param [String] project_name Name of project for artifact. Needed for OSX installs
+      # @param [String] artifact_url URL of the project install artifact
+      #
+      # @return nil
+      def install_artifact_on(host, artifact_url, project_name)
+        variant, _, _, _ = host[:platform].to_array
+        onhost_package_file = nil
+        if variant == 'eos'
+          host.get_remote_file( artifact_url )
+          onhost_package_file = File.basename( artifact_url )
+        elsif variant == 'solaris'
+          artifact_filename = File.basename( artifact_url )
+          artifact_folder = File.dirname( artifact_url )
+          fetch_http_file( artifact_folder, artifact_filename, '.' )
+          onhost_package_dir = host.tmpdir( 'puppet_installer' )
+          scp_to host, artifact_filename, onhost_package_dir
+          onhost_package_file = "#{ onhost_package_dir }/#{ artifact_filename }"
+        elsif variant == 'osx'
+          on host, "curl -O #{ artifact_url }"
+          onhost_package_file = "#{ project_name }*"
+        end
+
+        if variant == 'eos'
+          # TODO Will be refactored into {Beaker::Host#install_local_package}
+          #   immediately following this work. The release timing makes it
+          #   necessary to have this here separately for a short while
+          host.install_from_file( onhost_package_file )
+        elsif onhost_package_file
+          if onhost_package_dir
+            host.install_local_package( onhost_package_file, '.' )
+          else
+            host.install_local_package( onhost_package_file )
+          end
+        else
+          host.install_package( artifact_url )
```

Comment:
So this will give the whole artifact url as defined in the yaml file, yes? Like `http://builds.puppetlabs.lan/puppet-agent/1.9.1/artifacts/el/7/PC1/x86_64/puppet-agent-1.9.1-1.el7.x86_64.rpm` or `http://builds.puppetlabs.lan/puppet-agent/903bd6cbb8bfc6f88a6be61acc86b8bfa12b8690/artifacts/deb/xenial/PC1/puppet-agent_5.0.1.113.g903bd6c-1xenial_amd64.deb`?

I'm not 100% sure all package management sysmtes are going to be able to handle that as an argument, unless it's getting parsed somewhere that I've missed

```
root@plb4iyu3cgus7pk:~# apt-get install --force-yes -y http://builds.puppetlabs.lan/puppet-agent/903bd6cbb8bfc6f88a6be61acc86b8bfa12b8690/artifacts/deb/xenial/PC1/puppet-agent_5.0.1.113.g903bd6c-1xenial_amd64.deb
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package http://builds.puppetlabs.lan/puppet-agent/903bd6cbb8bfc6f88a6be61acc86b8bfa12b8690/artifacts/deb/xenial/PC1
E: Couldn't find any package by glob 'http://builds.puppetlabs.lan/puppet-agent/903bd6cbb8bfc6f88a6be61acc86b8bfa12b8690/artifacts/deb/xenial/PC1'
E: Couldn't find any package by regex 'http://builds.puppetlabs.lan/puppet-agent/903bd6cbb8bfc6f88a6be61acc86b8bfa12b8690/artifacts/deb/xenial/PC1'
```